samples/WebApi/README.md

```diff
@@ -0,0 +1,143 @@
+<add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+```
+
+Once you've added the package source, add a reference to the compiler by running the following command:
+
+`> dotnet add package Microsoft.DotNet.ILCompiler -v 1.0.0-alpha-* `
+
+## Add Core MVC services
+With the package successfully added to your project, your project's default registered MVC services must be modified.
+
+The default template's `AddMvc()` call registers a large set of middleware services by default, even if they are not needed by your application. This is not AOT-compilation friendly because it leads to large binaries and creates the risk of adding unsupported features.
+
+Open the file called `Startup.cs` and in the `ConfigureServices()` method and modify the line:
+
+```csharp
+services.AddMvc();
+```
+
+to
+
+```csharp
+services.AddMvcCore().AddJsonFormatters();
+```
+
+Replacing `AddMvc()` with `AddMvcCore()` adds only the basic MVC functionality. It's followed by explicit registration of services, which are required by the application - in this case the `JsonFormatter`. For more details see [Fabian Gosebrink's blog post comparing AddMvc and AddMvcCore](https://dzone.com/articles/the-difference-between-addmvc-and-addmvccore).
+
+## Using reflection 
+Runtime directives are XML configuration files, which specify which elements of your program are available for reflection. They are used at compile-time to enable AOT compilation in applications at runtime. 
+
+In this sample a basic rd.xml file has been added for a simple Web API application under the root project folder. Copy its contents to your application directory and modify the
+```xml <Assembly Name="SampleWebApi" Dynamic="Required All" /> ``` element to use your app's name.
+
+If your application makes use of reflection, you will need to create a rd.xml file specifying explicitly which assemblies and types should be made available. For example, in  your .NET Core Web API application, reflection is required to determine the correct namespace, from which to load the ``Startup`` type. Both are defined respectively via the `<Assembly>` and `<Type>` attributes. For example, in the case of our specific application;
+
+```xml 
+<Assembly Name="SampleWebApi">
+  <Type Name="SampleWebApi.Startup" Dynamic="Required All" />
+</Assembly>
+```
+
+At runtime, if a method or type is not found or cannot be loaded, an exception will be thrown. The exception message will contain information on the missing type reference, which you can then add to the rd.xml of your program.
+
+Once you've created a rd.xml file, navigate to the root directory of your project and open its `.csproj` file and in the first `<PropertyGroup>` element add the following:
+
+```xml
+<RdXmlFile>path_to_rdxml_file\rd.xml</RdXmlFile>
+```
+
+where path_to_rdxml_file is the location of the file on your disk.
+
+Directly below, add:
+
+```xml
+<RuntimeIdentifiers>runtime_identifier</RuntimeIdentifiers>
```

Comment:
RuntimeIdentifiers tag is not in the sample .csproj file in this PR. Is it really necessary?

Reply:
Not strictly. The RID needs to be available for `dotnet restore`, so the correct runtime package can be resolved. If the restore is run as part of `dotnet publish` the runtime is already supplied as an argument. 
I've left these as separate steps to make sure the package restore is explicitly called (in case a stale package is available in the cache).

Reply:
The issue with restore running as part of publish exists even without CoreRT for self-contained apps. I do not think we should be trying to rethink how publishing of standalone apps works here.

Reply:
E.g. Our instructions for publishing self-contained apps for Raspberry Pi does not worry about this either: https://github.com/dotnet/core/blob/master/samples/RaspberryPiInstructions.md

Reply:
Good point. I've reverted the instructions and .csproj to match the Raspberry Pi sample.